sdk/servicebus/service-bus/src/senderClient.ts

```diff
@@ -0,0 +1,210 @@
+import { isTokenCredential } from "@azure/core-amqp";
+import { ClientType } from "./client";
+import { generate_uuid } from "rhea-promise";
+import { ConnectionContext } from "./connectionContext";
+
+/**
+ * Client for sending messages to a topic or queue.
+ */
+export class ServiceBusSenderClient {
+  private _entityPath: string;
+  private _currentSender: Sender;
+  private _context: ConnectionContext;
+
+  /**
+   * Creates a ServiceBusClient for the Service Bus Namespace represented in the given connection
+   * string.
+   * @param entityConnectionString - Connection string of the form
+   * 'Endpoint=sb://my-servicebus-namespace.servicebus.windows.net/;SharedAccessKeyName=my-SA-name;SharedAccessKey=my-SA-key;EntityPath=new-queue'
+   * @param options Options to control ways to interact with the
+   * Service Bus Namespace.
+   * @returns ServiceBusClient
+   */
+  constructor(entityConnectionString: string, options?: ServiceBusClientOptions);
+
+  /**
+   * Creates a ServiceBusClient for the Service Bus Namespace represented in the given connection
+   * string.
+   * @param serviceBusConnectionString - Connection string of the form
+   * 'Endpoint=sb://my-servicebus-namespace.servicebus.windows.net/;SharedAccessKeyName=my-SA-name;SharedAccessKey=my-SA-key'
+   * @param options Options to control ways to interact with the
+   * Service Bus Namespace.
+   * @returns ServiceBusClient
+   */
+  constructor(
+    serviceBusConnectionString: string,
+    entityName: string,
+    options?: ServiceBusClientOptions
+  );
+
+  /**
+   * Instantiates a ServiceBusClient to interact with a Service Bus Namespace.
+   *
+   * @constructor
+   * @param host - The host name for the Service Bus namespace. This is likely to be similar to
+   * <yournamespace>.servicebus.windows.net
+   * @param credential - credential that implements the TokenCredential interface.
+   * @param options - Options to control ways to interact with the Service Bus
+   * Namespace.
+   */
+  constructor(
+    host: string,
+    entityName: string,
+    credential: TokenCredential,
+    options?: ServiceBusClientOptions
+  );
+
+  constructor(
+    hostOrConnectionString: string,
+    entityNameOrOptions?: string | ServiceBusClientOptions,
+    credentialOrServiceBusClientOptions?: TokenCredential | ServiceBusClientOptions,
+    options?: ServiceBusClientOptions
+  ) {
+    if (typeof entityNameOrOptions !== "string") {
+      // (entityConnectionString: string, options?: ServiceBusClientOptions)
+      const entityConnectionString = hostOrConnectionString;
+      const options = entityNameOrOptions;
+      // get the entity name from the connection string
+      const entityPathMatch = entityConnectionString.match(/^.+EntityPath=(.+?);{0,1}$/);
+
+      if (entityPathMatch!.length !== 2) {
+        throw new Error("Invalid entity connection string - no EntityPath");
+      } else {
+        this._entityPath = String(entityPathMatch![0]);
```

Comment:
I think the 0 index is the full match, but you actually want the 1st capture group
```suggestion
        this._entityPath = String(entityPathMatch![1]);
```

Reply:
You were correct. Removed this code and replaced with less busted code.